File: api/app/schemas/workspace.py
from pydantic import BaseModel, ConfigDict, Field, field_validator
from uuid import UUID
from typing import Optional, List

#############################################################################
#############################################################################
class WorkspaceCreate(BaseModel):
    name: str
    url: str
    slug: str

#############################################################################
#############################################################################
class WorkspaceUpdate(BaseModel):
    name: Optional[str] = None
    url: Optional[str] = None
    slug: Optional[str] = None

#############################################################################
#############################################################################
class WorkspaceResponse(BaseModel):
    id: UUID
    organization_id: UUID
    name: str
    url: str
    slug: str
    tags: List[str] = []

    model_config = ConfigDict(from_attributes=True)

    @field_validator("tags", mode="before")
    @classmethod
    def transform_ltree_to_str(cls, v):
        if v is None:
            return []
        return [str(tag) for tag in v]

File: dashboard/src/types/workspace.ts
export interface Workspace {
  id: string;
  organization_id: string;
  name: string;
  url: string;
  slug: string;
  tags: string[];
}

export interface WorkspaceCreate {
  name: string;
  url: string;
  slug: string;
}

export interface WorkspaceUpdate {
  name?: string;
  url?: string;
  slug?: string;
}